controller/controller.go

```diff
@@ -99,39 +100,111 @@ func (c *Controller) Attach(attachRequest map[string]string) resources.FlexVolum
+func (c *Controller) GetVolumeName(getVolumeNameRequest k8sresources.FlexVolumeGetVolumeNameRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
+	volumeName, ok := getVolumeNameRequest.Opts["volumeName"]
+	if !ok {
+		return k8sresources.FlexVolumeResponse{
+			Status:  "Failure",
+			Message: fmt.Sprintf("Failed checking volumeName, ", getVolumeNameRequest.Opts),
+		}
+
+	}
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume is attached",
+		VolumeName: volumeName,
+		Device:     volumeName,
+		Attached:   true,
+	}
+}
+
+//WaitForAttach Waits for a volume to get attached to the node
+func (c *Controller) WaitForAttach(waitForAttachRequest k8sresources.FlexVolumeWaitForAttachRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-waitForAttach-start")
+	defer c.logger.Println("controller-waitForAttach-end")
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume attached",
+		VolumeName: waitForAttachRequest.Name,
+		Device:     waitForAttachRequest.Name,
+		Attached:   true,
+	}
+}
+
+//IsAttached checks if volume is attached
+func (c *Controller) IsAttached(isAttachRequest k8sresources.FlexVolumeIsAttachedRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
+	return k8sresources.FlexVolumeResponse{
+		Status:     "Success",
+		Message:    "Volume is attached",
+		VolumeName: isAttachRequest.Name,
+		Device:     isAttachRequest.Name,
+		Attached:   true,
+	}
+}
+
 //Detach detaches the volume/ fileset from the pod
-func (c *Controller) Detach(detachRequest resources.FlexVolumeDetachRequest) resources.FlexVolumeResponse {
+func (c *Controller) Detach(detachRequest k8sresources.FlexVolumeDetachRequest) k8sresources.FlexVolumeResponse {
 	c.logger.Println("controller-detach-start")
 	defer c.logger.Println("controller-detach-end")
 
 	c.logger.Printf("detach-details %#v. (no operation done in the detach action)\n", detachRequest)
 
-	// TODO : no-op for now, will change with latest flex api update
+	// TODO : no-op for now, will change with latest k8sresources api update
 
-	return resources.FlexVolumeResponse{
+	return k8sresources.FlexVolumeResponse{
 		Status:  "Success",
 		Message: "Volume detached successfully",
 		Device:  detachRequest.Name,
 	}
 }
 
+//MountDevice mounts a device in a given location
+func (c *Controller) MountDevice(mountDeviceRequest k8sresources.FlexVolumeMountDeviceRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-MountDevice-start")
+	defer c.logger.Println("controller-MountDevice-end")
+	c.logger.Printf("mountDeviceRequest %#v", mountDeviceRequest)
+
+	mountRequest := k8sresources.FlexVolumeMountRequest{MountDevice: mountDeviceRequest.Name, MountPath: mountDeviceRequest.Path, Opts: mountDeviceRequest.Opts}
```

Comment:
so what is the diff between Mount vs MountDevice?
if u call here the mount function, are you sure k8s will not call it again in its mount flow? so u may end up with 2 mounts